apps/qubit/modules/settings/actions/diacriticsAction.class.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class SettingsDiacriticsAction extends SettingsEditAction
+{
+    // Arrays not allowed in class constants
+    public static $NAMES = [
+        'diacritics',
+    ];
+
+    public function earlyExecute()
+    {
+        parent::earlyExecute();
+
+        $this->updateMessage = $this->i18n->__('Diacritics settings saved.');
+
+        $this->settingDefaults = [
+            'diacritics' => 0,
+        ];
+    }
+
+    public function execute($request)
+    {
+        parent::execute($request);
+    }
+
+    public function uploadDiacritics($request)
+    {
+        $file = $request->getFiles('mappings');
+        $diacriticsMappingPath =
+            sfConfig::get('sf_plugins_dir') . DIRECTORY_SEPARATOR .
+            'arElasticSearchPlugin' . DIRECTORY_SEPARATOR .
+            'config' . DIRECTORY_SEPARATOR .
+            'diacritics_mapping.yml';
+
+        if (1 === (int) $request->getPostParameters()['diacritics']) {
+            if ('application/x-yaml' === $request->getFiles()['mappings']['type']) {
```

Comment:
I think you could set the mine type in the sfValidatorFile below. For example: https://github.com/artefactual/atom/blob/qa/2.x/apps/qubit/modules/informationobject/actions/uploadFindingAidAction.class.php#L54-L64.

Reply:
this didn't work for me, at least not the sfValidatorFile part... if i add that, it skips over processForm and uploadDiacritics. I've come up with a slightly different way of validation, though!

1 thing i wasn't sure how to do:
- how to not display the update message if an error message is being thrown... currently we get both :upside_down_face: